### PR TITLE
xcpMaxCto should be of bigger type, otherwise xcpMaxCto > XCPLOADER_P…

### PR DESCRIPTION
**Pull request by FlashForger**
…ACKET_SIZE_MAX(255) makes no sense

### CODE REVIEW

**Comment by feaser** [comment]
Thank you for your interest in the OpenBLT bootloader project and for taking the effort to issue the pull request. 

In essence you are correct. With macro XCPLOADER_PACKET_SIZE_MAX set to 255 (which is the default right now), the following check doesn't really make sense, if variable xcpMaxCto is 8-bit unsigned:
```
      if (xcpMaxCto > XCPLOADER_PACKET_SIZE_MAX)
```

The idea behind macro XCPLOADER_PACKET_SIZE_MAX is that you can lower it, in case you ever wanted to port and use the XcpLoader module on a microcontroller with RAM restrictions. In this case the above code does make sense and that's the reason why I would like to keep it like that.

Your suggestion of changing variable xcpMaxCto to 16-bit unsigned is an option. However, according to the XCP protocol, the CTO length can never be more than 255, so it's not really necessary.

In summary, you have a valid point and your feedback is much appreciated. If you take into account that by design, it is possible to reconfigure XCPLOADER_PACKET_SIZE_MAX to a value lower than 255, the code is fine as is. For this reason, I decided to not process your pull request.
